Ginger/Ginger/ALM/Repository/OctaneRepository.cs

```diff
@@ -75,27 +86,117 @@ public override List<string> GetTestPlanExplorer(string path)
 
         public override IEnumerable<object> GetTestSetExplorer(string path)
         {
-            throw new NotImplementedException();
+            return octaneCore.GetTestSetExplorer(path); 
         }
 
         public override object GetTSRunStatus(object tsItem)
         {
-            throw new NotImplementedException();
+            return octaneCore.GetTSRunStatus(tsItem);
         }
 
         public override void ImportALMTests(string importDestinationFolderPath)
         {
-            throw new NotImplementedException();
+            Reporter.ToLog(eLogLevel.DEBUG, "Start importing from QC");
+            //set path to import to               
+            if (importDestinationFolderPath == "")
+                importDestinationFolderPath = WorkSpace.Instance.Solution.BusinessFlowsMainFolder;
+
+            //show Test Lab browser for selecting the Test Set/s to import
+            QCTestLabExplorerPage win = new QCTestLabExplorerPage(QCTestLabExplorerPage.eExplorerTestLabPageUsageType.Import, importDestinationFolderPath);
+            win.ShowAsWindow(eWindowShowStyle.Dialog);
         }
 
         public override void ImportALMTestsById(string importDestinationFolderPath)
         {
             throw new NotImplementedException();
         }
 
-        public override bool ImportSelectedTests(string importDestinationPath, IEnumerable<object> selectedTests)
+        public override bool ImportSelectedTests(string importDestinationPath, IEnumerable<object> selectedTestSets)
         {
-            throw new NotImplementedException();
+            if (selectedTestSets != null && selectedTestSets.Count() > 0)
+            {
+                ObservableList<QCTestSetTreeItem> testSetsItemsToImport = new ObservableList<QCTestSetTreeItem>();
+                foreach (QCTestSetTreeItem testSetItem in selectedTestSets)
+                {
+                    //check if some of the Test Set was already imported                
+                    if (testSetItem.AlreadyImported == true)
+                    {
+                        Amdocs.Ginger.Common.eUserMsgSelection userSelection = Reporter.ToUser(eUserMsgKey.TestSetExists, testSetItem.TestSetName);
+                        if (userSelection == Amdocs.Ginger.Common.eUserMsgSelection.Yes)
+                        {
+                            //Delete the mapped BF
+                            File.Delete(testSetItem.MappedBusinessFlow.FileName);
+                            testSetsItemsToImport.Add(testSetItem);
+                        }
+                    }
+                    else
+                    {
+                        testSetsItemsToImport.Add(testSetItem);
+                    }
+                }
+
+                if (testSetsItemsToImport.Count == 0) return false; //noting to import
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6317725)